sycl/source/detail/graph_impl.hpp

```diff
@@ -72,6 +72,46 @@ class node_impl {
             std::unique_ptr<sycl::detail::CG> &&CommandGroup)
       : MCGType(CGType), MCommandGroup(std::move(CommandGroup)) {}
 
+  /// Tests if two nodes have the same content,
+  /// i.e. same command group
+  /// @param Node node to compare with
+  bool operator==(const node_impl &Node) {
+    if (MCGType != Node.MCGType)
+      return false;
+
+    if (MCGType == sycl::detail::CG::CGTYPE::Kernel) {
+      sycl::detail::CGExecKernel *ExecKernelA =
+          static_cast<sycl::detail::CGExecKernel *>(MCommandGroup.get());
+      sycl::detail::CGExecKernel *ExecKernelB =
+          static_cast<sycl::detail::CGExecKernel *>(Node.MCommandGroup.get());
+
+      if (ExecKernelA->MKernelName.compare(ExecKernelB->MKernelName) != 0)
+        return false;
+    }
+    if (MCGType == sycl::detail::CG::CGTYPE::CopyUSM) {
+      sycl::detail::CGCopyUSM *CopyA =
+          static_cast<sycl::detail::CGCopyUSM *>(MCommandGroup.get());
+      sycl::detail::CGCopyUSM *CopyB =
+          static_cast<sycl::detail::CGCopyUSM *>(MCommandGroup.get());
+      if ((CopyA->getSrc() != CopyB->getSrc()) ||
+          (CopyA->getDst() != CopyB->getDst()) ||
+          (CopyA->getLength() == CopyB->getLength()))
```

Comment:
Should this be `!=`? I.e.
```suggestion
          (CopyA->getLength() != CopyB->getLength()))
```
I made the corresponding changes in my other suggestion. If I am wrong, make sure to change it back there.

Reply:
Good catch. It was indeed a mistake in the initial code, fixed by your previous suggestion :+1: